lightning/restore/restore.go

```diff
@@ -858,11 +870,47 @@ func (rc *RestoreController) restoreTables(ctx context.Context) error {
 	wg.Wait()
 	close(stopPeriodicActions)
 
-	err := restoreErr.Get()
+	err = restoreErr.Get()
 	logTask.End(zap.ErrorLevel, err)
 	return err
 }
 
+func (rc *RestoreController) newChecksumManager() (ChecksumManager, error) {
+	pdAddr := rc.cfg.TiDB.PdAddr
+	pdVersion, err := common.FetchPDVersion(rc.tls, pdAddr)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
```

Comment:
fetch PD version will fail when using "TiDB backend" outside of the cluster (i.e. the pd-addr is inaccessible).

you should return a no-op manager for TiDB backend instead of an error.

Reply:
fixed, if checksum is not enabled, return nil instead